### PR TITLE
New Thumbnail Networking

### PR DESCRIPTION
**Pull request by Noztik**

### CODE REVIEW

**Comment by swampservers** [comment]
Is this just to deal with the 255 char limit?

I'm worried this may not work because entities aren't Valid on the clientside until the client can "see" them in PVS

**Comment by Noztik** [comment]
> 
> 
> Is this just to deal with the 255 char limit?
> 
> I'm worried this may not work because entities aren't Valid on the clientside until the client can "see" them in PVS

I've tested this on my own dedicated server and it doesn't throw any errors and works fine although I didn't really test the broadcast part when the entity wasn't in PVS
they should be valid regardless if they are in PVS or not anyway since the client can fetch them with ents.FindByClass("theater_thumbnail") anywhere

**Comment by swampservers** [comment]
> > Is this just to deal with the 255 char limit?
> > I'm worried this may not work because entities aren't Valid on the clientside until the client can "see" them in PVS
> 
> I've tested this on my own dedicated server and it doesn't throw any errors and works fine although I didn't really test the broadcast part when the entity wasn't in PVS
> they should be valid regardless if they are in PVS or not anyway since the client can fetch them with ents.FindByClass("theater_thumbnail") anywhere

that's not true, if the entity is out of PVS it won't be in the returned table before the client has seen it for the first time. please do a bit of testing to learn about how entity networking works

File: gamemodes/cinema/gamemode/modules/theater/sh_theater.lua [hunk 1]
```diff
@@ -283,8 +290,22 @@ if SERVER then
...
+			net.WriteEntity(self._ThumbEnt)
+			net.WriteString(self:VideoThumbnail())
+		net.Broadcast()
+		
+		ThumbnailTable = ThumbnailTable or {}
+		net.Receive("ThumbnailDelivery", function(len, ply)
+			local e = net.ReadEntity()
+			net.Start("ThumbnailDelivery")
+				net.WriteEntity(e)
+				net.WriteString(ThumbnailTable[e] or "")
```
Comment (swampservers) [inline gamemodes/cinema/gamemode/modules/theater/sh_theater.lua]: Should use entity.thumbnail on serverside too

**Pull request merged**